### PR TITLE
Add UI for referrals flow

### PR DESCRIPTION
**Pull request by Ferossgp**
Depends on #11055

### CODE REVIEW

**Comment by hesterbruikman** [comment]
Check ahead of release 1.6 which is to include advertiser flow for referral program only (based on new contract) plus only high-prio fixes that would otherwise go into a `1.5.1`. Therefore it is likely based of `1.5` instead of `develop`.

Does this PR include any dependencies on `develop` that are not in [1.5 branch](https://github.com/status-im/status-react/commits/release/1.5.x)?

**Review by hesterbruikman — APPROVED** [review]
As far as I can assess and reviewing Sergey's [test findings](https://notes.status.im/UBSE42EASKKHxHDjDgudJw#), this PR includes everything required for release 1.6 (Advertiser referrals flow only), hiding 1:1 friend invite flow UI (Referrer and Recipient) behind a feature flag). Note that the advertiser flow does not rely on contract interactions.

**Comment by flexsurfer** [comment]
@Ferossgp is this still wip? or can be reviewed ?

File: src/status_im/acquisition/core.cljs [hunk 1]
```diff
@@ -93,3 +100,8 @@
   {}
   [_]
   {::check-referrer nil})
+
+(re-frame/reg-sub
```
Comment (flexsurfer) [inline src/status_im/acquisition/core.cljs:119]: we still don't have decision regarding subs.cljs . so better to move it there, and use reg-root-subs

File: src/status_im/acquisition/gateway.cljs [hunk 2]
```diff
@@ -10,22 +10,31 @@
...
     (get acquisition-routes :registrations)))
 
 (fx/defn handle-error
   {:events [::on-error]}
-  [_ error]
-  {:utils/show-popup {:title   "Request failed"
+  [{:keys [db]} error]
+  {:db               (assoc-in db [:acquisition :network-status]
+                               (get network-statuses :error))
+   :utils/show-popup {:title   "Request failed"
```
Comment (flexsurfer) [inline src/status_im/acquisition/gateway.cljs]: i18n

File: src/status_im/acquisition/gateway.cljs [hunk 3]
```diff
@@ -36,29 +45,37 @@
...
+  {:http-get {:url        (get-url :clicks referrer)
+              :on-success (fn [response]
+                            (re-frame/dispatch (on-success (types/json->clj response))))
+              :on-error   (fn [response]
+                            (let [error (types/json->clj response)]
+                              (if (handled-error error)
+                                (handle-error error)
+                                (re-frame/dispatch [::on-error (:error error)]))))}})
+(re-frame/reg-sub
+ ::network-status
```
Comment (flexsurfer) [inline src/status_im/acquisition/gateway.cljs:80]: same here for subs.cljs

File: src/status_im/ui/components/invite/events.cljs [hunk 4]
```diff
@@ -74,8 +80,33 @@
   [_]
   {::terms-and-conditions nil})
 
-;; Invite reward
+(fx/defn redeem-success
+  {:events [::redeem-success]}
+  [{:keys [db]} account]
+  {:db (assoc-in db [:acquisition :accounts account :bonuses] 0)
+   ;; :utils/show-popup {:title   "Success"
```
Comment (flexsurfer) [inline src/status_im/ui/components/invite/events.cljs]: leftover?

File: src/status_im/ui/components/invite/modal.cljs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                        :height       20
+                        :margin-right 8}}]
+    [quo/text {:size   :small
+               :weight :medium}
+     (str value " ")]
+    [quo/text {:size   :small
+               :weight :medium}
+     name]]])
+
+(defn token-icon-style [idx]
```
Comment (flexsurfer) [inline src/status_im/ui/components/invite/modal.cljs]: usually, we move styles in separate styles.cljs but not mandatory i guess

**Comment by Ferossgp** [comment]
@flexsurfer it can be reviewed, it is WIP in the plan that it points to the old mainnet contract and I will need to rebase on #11055 for the new one

**Review by flexsurfer — APPROVED** [review]

**Comment by Serhy** [comment]
- Can create and see "Welcome" pop-up on Chats view:
- Accepting it did not sent Starter pack on Mainnet (but I received Starter pack on Rinkeby. Don't see any issue with that actually). 
- Declining the 'Share your IP for 7 days' dialog does not record  IP (I tested it by: Staying in the same mobile network -> create user from first click_id -> decline IP sharing (at this point no starter pack received in Rinkeby) -> flush the data in app - > create another user from the second click_id and Accept it: as result in Rinkeby Starter pack received

In the Invite friends view there are empty values and no icons (no issue on Rinkeby, - everything is present there):

Screenshot 2020-08-19 at 20 05 43

**Comment by Serhy** [comment]
Looks good to me! 
Above issue fixed. Advertiser flow works (tested also account creation from seed-phrase, new account, KeyCard account). 
Clicks counter upon referral link and number of attributions tracked accordingly in admin. No starter packs received by recipients (and by Advertisers) after successful recipient account creation.

**Comment by Serhy** [comment]
Note: Invite Friends is showing now the 'Referral bonus' view on Mainnet and Rinkeby. I suppose it's because of no "Disable 1-1 invite" flag is set.

**Comment by hesterbruikman** [comment]
> Note: Invite Friends is showing now the 'Referral bonus' view on Mainnet and Rinkeby. I suppose it's because of no "Disable 1-1 invite" flag is set.

Believe this is because the feature flag is set for `release` build. Can you confirm @Ferossgp ?

**Comment by Ferossgp** [comment]
Yes, it is hidden only for release, as in 1.5

**Comment by Serhy** [comment]
Then, let's merge, @Ferossgp !

**Pull request merged**